diff --git a/dist/VcIssuer.js b/dist/VcIssuer.js
index 976789a3560a89df59094c868239b6d468e08fff..8be2d2bb1b0fe914fe0abca5c9e0e7028b038e49 100644
--- a/dist/VcIssuer.js
+++ b/dist/VcIssuer.js
@@ -304,7 +304,7 @@ class VcIssuer {
             let preAuthorizedCode;
             let issuerState;
             try {
-                if (credentialRequest.format !== 'jwt_vc_json' && credentialRequest.format !== 'jwt_vc_json-ld' && credentialRequest.format !== 'vc+sd-jwt') {
+                if (credentialRequest.format !== 'jwt_vc_json' && credentialRequest.format !== 'jwt_vc_json-ld' && credentialRequest.format !== 'vc+sd-jwt' && credentialRequest.format !== 'ldp_vc') {
                     throw Error(`Format ${credentialRequest.format} not supported yet`);
                 }
                 else if (typeof this._jwtVerifyCallback !== 'function' && typeof jwtVerifyCallback !== 'function') {
