diff --git a/build/openid4vc-verifier/OpenId4VpVerifierService.js b/build/openid4vc-verifier/OpenId4VpVerifierService.js
index b4d58d511d21f3f474fd27792f1f1e971149ad54..98e739ba927fe260968ba3fa0260279fc14a8e66 100644
--- a/build/openid4vc-verifier/OpenId4VpVerifierService.js
+++ b/build/openid4vc-verifier/OpenId4VpVerifierService.js
@@ -157,6 +157,7 @@ let OpenId4VpVerifierService = class OpenId4VpVerifierService {
                     jwtSigner: jwtIssuer,
                     requestUri: hostedAuthorizationRequestUri,
                     expiresInSeconds: this.config.authorizationRequestExpiresInSeconds,
+                    additionalJwtPayload: options.additionalJwtPayload
                 }
                 : undefined,
             authorizationRequestPayload: requestParamsBase.response_mode === 'dc_api.jwt' || requestParamsBase.response_mode === 'dc_api'
