diff --git a/build/modules/vc/W3cCredentialsModule.js b/build/modules/vc/W3cCredentialsModule.js
index 05f1171469ed3b63dc2a9e1ded7ff99e7991a71b..ac2a424b8dd7cde8d9934c20c4d221877a2377b6 100644
--- a/build/modules/vc/W3cCredentialsModule.js
+++ b/build/modules/vc/W3cCredentialsModule.js
@@ -28,15 +28,15 @@ class W3cCredentialsModule {
         // Register the config
         dependencyManager.registerInstance(W3cCredentialsModuleConfig_1.W3cCredentialsModuleConfig, this.config);
         // Always register ed25519 signature suite
-        dependencyManager.registerInstance(SignatureSuiteRegistry_1.SignatureSuiteToken, {
-            suiteClass: signature_suites_1.Ed25519Signature2018,
-            proofType: 'Ed25519Signature2018',
-            verificationMethodTypes: [
-                dids_1.VERIFICATION_METHOD_TYPE_ED25519_VERIFICATION_KEY_2018,
-                dids_1.VERIFICATION_METHOD_TYPE_ED25519_VERIFICATION_KEY_2020,
-            ],
-            supportedPublicJwkTypes: [kms_1.Ed25519PublicJwk],
-        });
+        // dependencyManager.registerInstance(SignatureSuiteRegistry_1.SignatureSuiteToken, {
+        //     suiteClass: signature_suites_1.Ed25519Signature2018,
+        //     proofType: 'Ed25519Signature2018',
+        //     verificationMethodTypes: [
+        //         dids_1.VERIFICATION_METHOD_TYPE_ED25519_VERIFICATION_KEY_2018,
+        //         dids_1.VERIFICATION_METHOD_TYPE_ED25519_VERIFICATION_KEY_2020,
+        //     ],
+        //     supportedPublicJwkTypes: [kms_1.Ed25519PublicJwk],
+        // });
         dependencyManager.registerInstance(SignatureSuiteRegistry_1.SignatureSuiteToken, {
             suiteClass: signature_suites_1.Ed25519Signature2020,
             proofType: 'Ed25519Signature2020',
