diff --git a/dist/authorization-response/PresentationExchange.js b/dist/authorization-response/PresentationExchange.js
index 547a02a7b58e26e0cd8d6b4398abc1fbb456af01..8dfca1c2ae5252e97b3742d764a87cda2c845bbd 100644
--- a/dist/authorization-response/PresentationExchange.js
+++ b/dist/authorization-response/PresentationExchange.js
@@ -272,15 +272,16 @@ class PresentationExchange {
             if (verifyPresentationCallback && evaluationResults.value !== undefined) {
                 // Verify the signature of all VPs
                 yield Promise.all(presentationsToVerify.map((presentation) => __awaiter(this, void 0, void 0, function* () {
+                    let verificationResult
                     try {
-                        const verificationResult = yield verifyPresentationCallback(presentation, evaluationResults.value);
-                        if (!verificationResult.verified) {
-                            throw new Error(types_1.SIOPErrors.VERIFIABLE_PRESENTATION_SIGNATURE_NOT_VALID + (verificationResult.reason ? `. ${verificationResult.reason}` : ''));
-                        }
+                        verificationResult = yield verifyPresentationCallback(presentation, evaluationResults.value);
                     }
                     catch (error) {
                         throw new Error(types_1.SIOPErrors.VERIFIABLE_PRESENTATION_SIGNATURE_NOT_VALID);
                     }
+                    if (!verificationResult.verified) {
+                        throw new Error(types_1.SIOPErrors.VERIFIABLE_PRESENTATION_SIGNATURE_NOT_VALID + (verificationResult.reason ? `. ${verificationResult.reason}` : ''));
+                    }
                 })));
             }
             PresentationExchange.assertValidPresentationSubmission(evaluationResults.value);
